uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-comments-tab.component.spec.js

```diff
@@ -123,6 +126,56 @@ define([
       });
     });
 
+    describe('can submit', function () {
+      describe('when the comment box has text', function () {
+        beforeEach(function () {
+          controller.comment.text = 'Request comment';
+        });
+
+        it('allows the request to be submitted', function () {
+          expect(controller.canSubmit()).toBe(true);
+        });
+      });
+
+      describe('when the comment box does not have text', function () {
+        beforeEach(function () {
+          controller.comment.text = '';
+        });
+
+        it('does not allow the request to be submitted', function () {
+          expect(controller.canSubmit()).toBe(false);
+        });
+      });
+    });
+
```

Comment:
Need a test what happens you click 'add comment', then submit.

Reply:
That test is added here:

https://github.com/civicrm/civihr/pull/2474/files/cbb0c6d0fa31d2ad291e37200e66ee56fec53d9e#diff-ffb401af4f8024160426f6f84bc38de5R140

Since when you add a comment the text box is cleared (just like in the beforeEach for this test).

I could add that test, but it would be redundant ?!?!?